contribute/linux/development_tree.rst

```diff
@@ -0,0 +1,161 @@
+the community. SOF Linux maintainers will on a regular basis,
+typically weekly, merge all upstream contributions into the SOF tree.
+
+
+Development flow
+****************
+
+The SOF Linux maintainers are
+
++---------------+-------------------+---------------+
+| Intel	        | Pierre Bossart    | @plbossart    |
++---------------+-------------------+---------------+
+| Intel         | Ranjani Sridharan | @ranj063      |
++---------------+-------------------+---------------+
+| Intel         | Kai Vehmanenaran  | @kv2019i      |
++---------------+-------------------+---------------+
+| NXP           | Daniel Baluta     | @dbaluta      |
++---------------+-------------------+---------------+
+
+The SOF maintainers apply the following process
+
+1. Mirror all SOF patches to topic/sof-dev-rebase
+*************************************************
+
+This mirroring consists in doing a set of "git cherry-pick" operations
+from topic/sof-dev to topic/sof-dev-rebase. Once all development
+patches are applied, SOF maintainers will add the relevant
+Signed-off-by and Reviewed-by 'tags'.
+
+In specific cases, incremental patches will be squashed to simplify
+upstream reviews, and the other of patches changed, but in all cases
+the intent is that the two topic/sof-dev and topic/sof-dev-rebase
+provide the same code (as seen with git diff or diff -r).
+
+2. Upstream merge/rebase
+************************
+
+When the two branches are integrated, the SOF maintainer will create
+an upstream baseline. This baseline is then merged locally on top of
+topic/sof-dev, then pushed as a dedicated PR and run through the CI
+tests. The merge may in some cases create conflicts that have to be
+resolved locally by the maintainer. Once the PR is deemed suitable for
+integration, the maintainer will use a 'Commit merge' operation (in
+contrast to the 'rebase-and-merge' used for development).
+
+In parallel, the topic/sof-dev-rebase branch is rebased on top of the
+same baseline, and again compared to the topic/sof-dev branch. After
+the two separate operations of merge and rebase on the two branches,
+these two branches should again be identical. The next effect of the
+rebase is that all patches already integrated by ALSA/ASoC maintainers
+'disappear'. In other words the topic/sof-dev-rebase branch helps
+visualize the all patches not currently merged.
+
+3. Upstream contributions
+*************************
+
+The SOF maintainer will generate patch sets and send them with a cover
+to the alsa-devel mailing list, with the maintainers in Cc:. In most
+cases the patches are approved without issues, but the ALSA/ASoC
+maintainers or members of the community may provide feedback and
+request some changes. In those cases, the changes are applied on
+topic/sof-dev, will then be mirrored and squashed on
+topc/sof-dev-rebase, and submitted again. Under no circumstances
+should the SOF maintainer handle changes to the topic/sof-dev-rebase
+directly.
+
+4. Exceptions
+*************
```

Comment:
Anything for ABI alignment as an exception ?

Reply:
no, why? Someone adds a patch on sof-dev, and we merge it. we should never have someone send ABI-related patches and bypass the GitHub integration, that would be unmanageable.

Reply:
Sorry, I was meaning that there may be some delay or further discussion as a result of an ABI change. i.e. it wont be automatically merged if same ABI is not aligned with FW and vice versa ? Or do you think this would benefit from a different document altogether ?

Reply:
Ah yes, I was only talking about driver patches. When there is an interaction with the firmware, it's indeed a different process, but that is a discussion between the two sides of SOF, it doesn't involved maintainers on the ASoC/ALSA/Linux side.

I'll let you write the process for ABI changes :-)

Reply:
hehe, ok - it's on my TODO list :)